### ISSUE TITLE
Triggers not working in game

### ISSUE THREAD

**Issue by MovablePizza** [HipsterSloth/PSMoveSteamVRBridge#69]
Some games work, for example The Lab, Belko Experiment etc.

But in Beat Saber I can't use the triggers whatsoever, same goes for Google Earth.

I can't get past their home screens simply because I have no way of clicking, any help?

These are my steamVR settings:
   "LastKnown" : {
      "HMDManufacturer" : "Oculus",
      "HMDModel" : "Oculus Rift DK2"
   },
   "direct_mode" : {
      "count" : 4,
      "edidVid3" : 53822
{
  "driver_vridge": {
    "fakeSensorEnabled": false,
    "renderScale": 100,
    "windowHeight": 1440,
    "windowWidth": 2560
  },
  "psmove": {
    "circle": "a",
    "cross": "button_8",
    "meters_per_touchpad_units": 0.075000002980232239,
    "move": "touchpad",
    "ps": "system",
    "rumble_suppressed": true,
    "select": "button_9",
    "square": "grip",
    "start": "button_10",
    "triangle": "application_menu",
    "trigger": "trigger",
    "use_spatial_offset_after_touchpad_press_as_touchpad_axis": true
  },
  "psmove_settings": {
    "psmove_filter_hmd_serial": "04:76:6e:c0:8b:d6"
  },
  "steamvr": {
    "activateMultipleDrivers": true,
    "defaultMirrorView": 1,
    "mirrorViewGeometry": "1249 473 1240 1165",
    "renderTargetMultiplier": 1.7000000476837158
  }
}

**Comment by MovablePizza** [comment]
I've found that in several other games it happens as well, unlike Beat Saber (where my controllers show up but don't do anything) they just don't show up in games like Google Earth, Waltz of the wizard, NVIDIA Funhouse etc.

It's because the games aren't launching through SteamVR, they're opening as independent applications. I can't access the SteamVR overlay in them, so they have no controllers.

Does anyone know a fix for this or am I just doomed to have a 50/50 chance of any game working?

**Comment by HipsterSloth** [comment]
The default PSMove Controller bindings might not be working in those games. You might need to set up per-game controller bindings using the SteamVR button binding tool:

https://github.com/HipsterSloth/PSMoveSteamVRBridge/wiki/Steam-VR-Setup#per-game-input-bindings

I assume you are running version 1.6.0 of PSMoveSteamVRBridge and the latest version of SteamVR (1.1.4), non-beta? If so the controller bindings you posted above from steamvr.vrsettings are no longer used. This is because the latest version of PSMoveSteamVRBridge upgraded to the new SteamVR input API.

**Comment by lilcoconut20** [comment]
Tried different trigger bindings and using default bindings. None of the buttons work in beat saber. I did a clean install for SteamVr(1.1.4) and beat saber with mods and no mods. I remember trying PSMoveSteamVRBridge 1.6.0 when it first released and SteamVr(1.1.3) the triggers worked but not the application menu button(triangle).

These are the default binding for trigger
![capture](https://user-images.githubusercontent.com/45697621/49897257-77edc780-fe0a-11e8-944a-3897968198ec.JPG)
![capture2](https://user-images.githubusercontent.com/45697621/49897264-7c19e500-fe0a-11e8-8b96-a422b9119b8f.JPG)

**Comment by MovablePizza** [comment]
Yeah I tried that as well, if any fix comes let me know. I ended up refunding and swapping it for Job Simulator. If it can be fixed for beat saber I'd love to rebuy it.

**Comment by Gaspadlo** [comment]
So i am not the only one... This has been driving me crazy for last couple of days, thinking that i have to be doing something wrong...

**Comment by Slanger29** [comment]
i only have psmove to play beat saber :(
having the same issue, looking for solution

**Comment by Cretkensigh** [comment]
when you start steamvr home does it show oculus touch or vive controllers on the wall screen? if its showing touch controller you need to make it emulate vive because the oculus has capacitive buttons  meaning they read not just when pressed but when you have a finger on them ot not and the move controllers cannot do this.

i found this when trying arizona sunshine and couldnt pickup/activate anything. was emulating a oculus with dk1 bit once i switched to vive emulation it worked fine. hopw this helps

**Comment by pupphelper** [comment]
I am also having the same problem.  The Triggers work fine in SteamVR home, as well as some other games.  But in beat saber, I can't use the trigger to start the game.  I tried going back to 1.51 as the OP mentioned, but that also didn't work for me.   I'm using PSMoveSteamVRBridge with ALVR (controller disabled) with 2 PSmove controllers.

**Comment by timmiles13** [comment]
Have a same issue, waiting for solution too. All the same, as pupphelper said.

**Comment by ahmadsyami** [comment]
any solution for this problem? just bought beat saber today sad cant play

**Comment by Cretkensigh** [comment]
has any body tried what i was talking about? dont seem to get a yey or ney with these issues. it worked for me and my issue was exaclty as you describe only about a year ago lol

**Comment by Zoradius** [comment]
So I was able to use a DS4 controller connected via Bluetooth for that trigger input, but sadly the game was still unplayable for me in terms of motion tracking. (Well that solution is anything but elegant.)

**Comment by Lrob82** [comment]
I’m having the same problem. When I’ve checked the buttons on both move controllers  are showing on pc as the same controller.

**Comment by Team-on** [comment]
Find solution to configurtate PsMove for any game.
https://steamcommunity.com/sharedfiles/filedetails/?id=1545634111

**Comment by redcubie** [comment]
@Team-on I have seen this article already but you bringing it up made me go over it again for something I missed and lo and behold there it was, the secret to make simulated triggers, which I missed when I glossed over it previously. Thanks!

**Comment by manobroda** [comment]
Same problem in Beat Saber. When I start SteamVR it shows the usual PSMove Controller, and it works fine. But in Beat Saber, only the walk/take button works well, and it is useless in this game.
Any other ideas??

**Comment by manobroda** [comment]
I just founded that PSMoveSteamVRBridge  Release 16 "moved configuration out of steamvr.vrsettings into %appdata%/PSMoveSteamVRBridge".

In this path, there are psmove_[MAC].json files. Inside it there is a key bindings section, like steamvr.vrsetting.

In my case, only "move" have a key binded, that corresponde exactly wrich button is working.

But I still do not know wrich key name I have to replace in "none".

Any ideas, or help??
